### ISSUE TITLE
SMHP: to document how to disk setup on head-node for docker and enroot 

### ISSUE THREAD

**Issue by verdimrc** [aws-samples/awsome-distributed-training#127]
Head node is provisioned with 100 GB EBS, barely enough to docker pull a Pytorch container (23+GB). Remaining 16GB, insufficient for enroot to proceed.

To document the steps needed:
- Assume FSx Lustre mounted to `/fsx`.
- Before cluster creation: update lcc enroot.conf, change `ENROOT_CACHE_PATH` from `/opt/enroot` to `/fsx/enroot`. 
- Create-cluster. Make sure that controller instance type has sufficiently large NVMe (e.g., `m5d.8xlarge` or larger) to hold enroot temporary files.
- Go to the controller node, edit `enroot.conf` to make sure it contains:
  ```text
  ENROOT_RUNTIME_PATH        /opt/dlami/nvme/tmp/enroot/user-$(id -u)
  ENROOT_CONFIG_PATH         ${HOME}/enroot
  ENROOT_CACHE_PATH          /fsx/enroot
  ENROOT_DATA_PATH           /opt/dlami/nvme/tmp/enroot/data/user-$(id -u)
  ENROOT_TEMP_PATH           /opt/dlami/nvme/tmp
  ```

  As precautionary measure (need to verify needed or not), precreate the directories:

  ```bash
  sudo mkdir -p /opt/dlami/nvme/tmp/enroot/
  sudo chmod 1777 /opt/dlami/nvme/tmp/enroot/

  sudo mkdir -p /opt/dlami/nvme/tmp/enroot/data/
  sudo chmod 1777 /opt/dlami/nvme/tmp/enroot/data/

  sudo mkdir -p /opt/dlami/nvme/tmp
  sudo chmod 1777 /opt/dlami/nvme/tmp

  sudo mkdir -p /fsx/enroot
  sudo chmod 1777 /fsx/enroot
  ```

For historical context, I tried to set `ENROOT_TEMP_PATH` to `/fsx/..`, but the `enroot import ...` will hang for more than 15minutes, then raise `Unrecognised xattr prefix lustre.lov` errors.

**Comment by verdimrc** [comment]
Update: this is not the right fix still.

**Comment by sean-smith** [comment]
You also need to relocate docker cache as well. Enroot uses docker layers when you run an enroot import. That looks like:
```
# switch docker to use local nvme storage
sudo su
cat <<EOL >> /etc/docker/daemon.json
{
    "data-root": "/opt/dlami/nvme"
}
EOL
systemctl restart docker
exit
```

**Comment by verdimrc** [comment]
The steps to relocate paths can be applied to compute nodes too (p5 has nearly 7TB NVMe), as compute nodes mount /tmp similar to how head node does.

**Comment by verdimrc** [comment]
To clarify: instead of applied to the controller node, the solution should be applied to compute instances (p4, p5) since they're the ones who have NVMe mounted to `/opt/dlami/nvme`. Once applied, users should use compute nodes to download or create images.

**Comment by verdimrc** [comment]
Closed via #130
